src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1374,21 +1374,32 @@
         WHERE f.value IS NULL AND fv.value IS NULL")
 
   (log/info (trs "[5/8] Cleaning up duplicate null values..."))
-  (let [json-null-value-id (some-> (jdbc/query-to-vec "select id from fact_values
-                                                       where value_type_id = 5 and value = 'null'")
-                                   first
-                                   :id)
-        real-null-value-id (some-> (jdbc/query-to-vec "select id from fact_values
-                                                       where value_type_id = 4 and value is null")
-                                   first
-                                   :id)]
-    (if (and json-null-value-id real-null-value-id)
-      (do
-        (jdbc/do-prepared "update facts_transform set fact_value_id = (?) where fact_value_id = (?)"
-                          [real-null-value-id json-null-value-id])
-        (jdbc/do-prepared "delete from fact_values where id = (?)"
-                          [json-null-value-id]))
-      (log/info (trs "  ... none found"))))
+  ;; only do this if the DB has some null values
+  (when (pos? (-> (jdbc/query-to-vec "select count(*) from fact_values
+                                      where value_type_id in (4,5)
+                                      and value in (null, 'null')")
```

Comment:
I think this won't work due to null comparison issues

```
puppetdb=# select * from fact_values where value_integer in (null);                                                                                                                                                                           
 id | value_hash | value_type_id | value_integer | value_float | value_string | value_boolean | value                                                                                                                                         
----+------------+---------------+---------------+-------------+--------------+---------------+-------                                                                                                                                        
(0 rows)                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                              
Time: 0.626 ms
```

Reply:
ARGH good catch